src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -282,11 +283,9 @@ ucs_status_t uct_ib_mlx5_txwq_init(uct_worker_h worker, uct_ib_mlx5_txwq_t *txwq
         return UCS_ERR_IO_ERROR;
     }
 
-    if ((qp_info.bf.size == 0) || !ucs_is_pow2(qp_info.bf.size) ||
-        (qp_info.sq.stride != MLX5_SEND_WQE_BB) ||
-        (qp_info.bf.size != UCT_IB_MLX5_BF_REG_SIZE) ||
-        !ucs_is_pow2(qp_info.sq.wqe_cnt))
-    {
+    if ((qp_info.sq.stride != MLX5_SEND_WQE_BB) || !ucs_is_pow2(qp_info.sq.wqe_cnt) ||
+        ((qp_info.bf.size != 0) && (!ucs_is_pow2(qp_info.bf.size) ||
+        (qp_info.bf.size != UCT_IB_MLX5_BF_REG_SIZE)))) {
```

Comment:
i think we can drop !ucs_is_pow2(qp_info.bf.size) check, right? since UCT_IB_MLX5_BF_REG_SIZE is a power of 2